mos_tests/neutron/python_tests/test_dvr.py

```diff
@@ -387,3 +391,343 @@ def test_routing_east_west(self, prepare_openstack):
+            4.  Check that creation isn't available
+            5.  Create router with parameter Distributed = False
+            6.  Check that creation isn't available
+            7.  Create router without this parameter
+            8.  Log in as admin user
+            9.  Check that parameter Distributed is true
+        """
+
+        # Find the admin tenant
+        admin_role = self.os_conn.keystone.roles.find(name='admin')
+        admin_tenant = None
+        for tenant in self.os_conn.keystone.tenants.list():
+            if admin_role in tenant.manager.role_manager.list():
+                admin_tenant = tenant
+                break
+        assert admin_tenant, "Can't find the tenant with admin role"
+
+        # Create new user
+        # Member role is used by default
+        username = 'test_dvr'
+        userpass = 'test_dvr'
+        # But at first check if the same user exist
+        # try to find it and delete
+        try:
+            user = self.os_conn.keystone.users.find(name=username)
+            self.os_conn.keystone.users.delete(user)
+        except Exception as e:
+            logger.info('Tried to clean up user with result: {}'.format(e))
+        # Actual user creation is here
+        user = self.os_conn.keystone.users.create(name=username,
+                                                  password=userpass,
+                                                  tenant_id=admin_tenant.id)
+
+        # Find the certificate for the current env
+        # and log in with new user in new netron client
+        cert = self.env.certificate
+        path_to_cert = None
+        if cert:
+            with NamedTemporaryFile(prefix="fuel_cert_", suffix=".pem",
+                                    delete=False) as f:
+                f.write(cert)
+            path_to_cert = f.name
+
+        auth_url = self.os_conn.keystone.auth_url
+        tenant_name = self.os_conn.keystone.project_name
+        neutron = neutronclient.Client(username=username,
+                                       password=userpass,
+                                       tenant_name=tenant_name,
+                                       auth_url=auth_url,
+                                       ca_cert=path_to_cert)
+
+        # Try to create router with explicit distributed True value
+        # by user with memeber role but in admin tenant
+        with pytest.raises(NeutronClientException) as e:
+            router = {'name': 'router01', 'distributed': True}
+            router_id = neutron.create_router(
+                            {'router': router})['router']['id']
+        allowed_msg = 'disallowed by policy'
+        err_msg = 'Failed to create the router, exception: {}'.format(e)
+        assert allowed_msg in str(e.value), err_msg
+
+        # Try to create router with explicit distributed False value
+        # by user with memeber role but in admin tenant
+        with pytest.raises(NeutronClientException) as e:
```

Comment:
it is not clear how we use 

pytest.raises(NeutronClientException) as e

here